src/components/application_manager/test/commands/mobile/on_system_request_notification_test.cc

```diff
@@ -184,6 +184,52 @@ TEST_F(OnSystemRequestNotificationTest, DISABLED_Run_ProprietaryType_SUCCESS) {
 #endif  // EXTENDED_POLICY
 }
 
+TEST_F(OnSystemRequestNotificationTest, Run_EmptyMessage_UNSUCCESS) {
+  const RequestType::eType kRequestType = RequestType::PROPRIETARY;
+
+  MessageSharedPtr msg = CreateMessage();
+  (*msg)[strings::params][strings::connection_key] = kConnectionKey;
+  (*msg)[strings::msg_params][strings::request_type] = kRequestType;
+
+  SharedPtr<OnSystemRequestNotification> command =
+      CreateCommand<OnSystemRequestNotification>(msg);
+
+  MockAppPtr mock_app = CreateMockApp();
+  EXPECT_CALL(mock_app_manager_, application(kConnectionKey))
+      .WillOnce(Return(mock_app));
+  MockPolicyHandlerInterface mock_policy_handler;
+  EXPECT_CALL(mock_app_manager_, GetPolicyHandler())
+      .WillRepeatedly(ReturnRef(mock_policy_handler));
+  std::string policy_app_id;
+  EXPECT_CALL(*mock_app, policy_app_id()).WillOnce(Return(policy_app_id));
+  EXPECT_CALL(mock_policy_handler, IsRequestTypeAllowed(_, _))
+      .WillOnce(Return(true));
+
+#ifdef EXTENDED_POLICY
+  EXPECT_CALL(mock_app_manager_, GetPolicyHandler())
```

Comment:
@okozlovlux please use On_CALL for such getters, and I would refer to setup in Setup